### PR TITLE
board_helper: support dmic01 and dmic16k link

### PR DESCRIPTION
**Pull request by brentlu**
Add support functions for machine drivers to initialize dmic01 and dmic16k DAI link structure.

### CODE REVIEW

File: sound/soc/intel/boards/sof_board_helpers.h [hunk 1]
```diff
@@ -47,6 +49,10 @@ struct sof_card_private {
 
 int sof_intel_board_card_late_probe(struct snd_soc_card *card);
 
+int sof_intel_board_set_dmic01_link(struct device *dev,
+				    struct snd_soc_dai_link *link, int be_id);
+int sof_intel_board_set_dmic16k_link(struct device *dev,
+				     struct snd_soc_dai_link *link, int be_id);
```
Comment (bardliao) [inline sound/soc/intel/boards/sof_board_helpers.h]: Can we combine sof_intel_board_set_dmic16k_link and sof_intel_board_set_dmic01_link?

Comment (plbossart) [reply sound/soc/intel/boards/sof_board_helpers.h]: Not sure what you mean by "combine" @bardliao ? Maybe we can use the same helper but the dailink name and dai name need to be different for backwards compatibility.

Comment (bardliao) [reply sound/soc/intel/boards/sof_board_helpers.h]: > Not sure what you mean by "combine" @bardliao ? Maybe we can use the same helper but the dailink name and dai name need to be different for backwards compatibility.

I mean use the same helper, just like @brentlu does now.

**Review by bardliao — DISMISSED** [review]

File: sound/soc/intel/boards/sof_board_helpers.h [hunk 2]
```diff
@@ -38,15 +39,24 @@ struct sof_card_private {
 	enum sof_ssp_codec codec_type;
 	enum sof_ssp_codec amp_type;
 
+	int dmic_num;
```
Comment (plbossart) [inline sound/soc/intel/boards/sof_board_helpers.h]: this should be dmic_be_num, this has nothing to do with the number of mics.

File: sound/soc/intel/boards/sof_cs42l42.c [hunk 3]
```diff
@@ -510,10 +453,10 @@ static int sof_audio_probe(struct platform_device *pdev)
 	ctx->amp_type = sof_ssp_detect_amp_type(&pdev->dev);
 
 	if (soc_intel_is_glk()) {
-		dmic_be_num = 1;
+		ctx->dmic_num = 1;
```
Comment (plbossart) [inline sound/soc/intel/boards/sof_cs42l42.c]: why not use ````ctx->dmic_be_num````

**Review by plbossart — CHANGES_REQUESTED** [review]
naming nit-pick below to avoid confusions.

File: sound/soc/intel/boards/sof_cs42l42.c (hunk 3 above)
Comment (brentlu) [reply sound/soc/intel/boards/sof_cs42l42.c]: sure, no problem

**Review by plbossart — APPROVED** [review]
Nice cleanup @brentlu

**Review by bardliao — APPROVED** [review]

**Comment by plbossart** [comment]
Again a problem with CML_SKU0955_HDA in https://sof-ci.01.org/linuxpr/PR4636/build426/devicetest/index.html

this happened in https://sof-ci.01.org/linuxpr/PR4641/build414/devicetest/index.html

@keqiaozhang @fredoh9 is this device working?

**Pull request merged**